src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/vacation-requests-table-toolbar.tsx

```diff
@@ -0,0 +1,168 @@
+import { Add, Cancel, Edit } from "@mui/icons-material";
+import { Box, Collapse, Grid, styled } from "@mui/material";
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import DeleteVacationRequests from "../../delete-vacation-requests";
+import ToolbarForm from "./toolbar-form/toolbar-form";
+import DeleteVacationRequestStatuses from "../../delete-vacation-request-statuses";
+import { VacationRequest, VacationRequestStatus } from "../../../../../generated/client";
+import { GridRowId } from "@mui/x-data-grid";
+import { DataGridRow, ToolbarFormModes } from "../../../../../types";
+import ToolbarDeleteButton from "./toolbar-delete-button";
+import ToolbarTitle from "./toolbar-title";
+import FormToggleButton from "./toolbar-form-toggle-button";
+import ConfirmationHandler from "../../../../contexts/confirmation-handler";
+
+/**
+ * Component properties
+ */
+interface Props {
+  vacationRequests: VacationRequest[];
+  setVacationRequests: Dispatch<SetStateAction<VacationRequest[]>>;
+  vacationRequestStatuses: VacationRequestStatus[];
+  setVacationRequestStatuses: Dispatch<SetStateAction<VacationRequestStatus[]>>;
+  setFormOpen: Dispatch<SetStateAction<boolean>>;
+  formOpen: boolean;
+  selectedRowIds: GridRowId[] | undefined;
+  rows: DataGridRow[];
+}
+
+/**
+ * Table toolbar component
+ *
+ * @param props component properties
+ */
+const TableToolbar = ({
+  vacationRequests,
+  setVacationRequests,
+  vacationRequestStatuses,
+  setVacationRequestStatuses,
+  setFormOpen,
+  formOpen,
+  selectedRowIds,
+  rows
+}: Props) => {
+  const [toolbarOpen, setToolbarOpen] = useState<boolean>(false);
```

Comment:
Type can be inferred here, no need to declare the type

Reply:
Inferred type